lightning/src/ln/functional_test_utils.rs

```diff
@@ -97,6 +97,16 @@ impl<'a, 'b, 'c> Drop for Node<'a, 'b, 'c> {
 			assert!(self.node.get_and_clear_pending_events().is_empty());
 			assert!(self.chan_monitor.added_monitors.lock().unwrap().is_empty());
 
+			// Check that if we serialize the Router, we can deserialize it again.
+			{
+				let mut w = test_utils::TestVecWriter(Vec::new());
+				self.router.write(&mut w).unwrap();
+				let _ = Router::read(&mut ::std::io::Cursor::new(&w.0), RouterReadArgs {
+					chain_monitor: Arc::clone(&self.chain_monitor) as Arc<chaininterface::ChainWatchInterface>,
+					logger: Arc::clone(&self.logger) as Arc<Logger>
+				}).unwrap();
```

Comment:
Any way to go further and test old_router against deserialized_router to check we don't loss any data?